guacamole-ext/src/main/java/org/apache/guacamole/net/auth/simple/SimpleConnection.java

```diff
@@ -98,6 +99,10 @@ public void setAttributes(Map<String, String> attributes) {
     public GuacamoleTunnel connect(GuacamoleClientInformation info)
             throws GuacamoleException {
 
+        // Filter in connection-time information
+        TokenFilter tokenFilter = new TokenFilter();
+        tokenFilter.filterPrompts(config.getParameters(),info.getParameters());
```

Comment:
Yeah, I think I see your point, here.  Will see what options there are to take care of this.

Reply:
i think you should save the prompts per each configuration, and maybe calculate them in advance

Reply:
I thought about that, and one of my earlier implementation attempts actually did function this way.  The reason I opted for this sort of inline processing method was to be consistent with the way other tokens are processed, which is done at connection time rather than being done ahead of time.  I think this is probably mostly due to the fact that the GUAC_DATE and GUAC_TIME tokens are expected to be processed in real-time so that their values are current.

That said, the prompts don't necessarily have to work the same as the other tokens if there's a compelling reason to do something else.  I went back and forth on this a lot, and eventually ended up at this point.

I think for this particular issue you raise I'll just make a copy of the config object and filter that one for prompts, and it can get discarded after the connection is finished.

Reply:
Okay, I'm now copying the config in this method, so that the values are placed into that non-persistent config, which will then get discarded after the connection finishes.